src/operator/gitea.ts

```diff
@@ -524,6 +612,110 @@ async function createReposAndAddToTeam(
     )
 }
 
+// Logic to create a webhook for repos in a organization
+export async function createBuildWebHook(
+  repoApi: RepositoryApi,
+  teamName: string,
+  buildWorkspace: { buildName: string; repoUrl: string },
+) {
+  try {
+    const repoName = buildWorkspace.repoUrl.split('/').pop()!
```

Comment:
I think you can make a function out of this one and reuse it:
```
function getRepoNameFromUrl(url: string): string | null {
  const parts = url.split('/')
  return parts.length ? parts.pop() || null : null
}
```
```suggestion
    const repoName = getRepoNameFromUrl(buildWorkspace.repoUrl)
```